scripts/build/tests.js

```diff
@@ -167,23 +156,26 @@ exports.cleanTestDirs = cleanTestDirs;
  * @param {boolean} [keepFailed]
  * @param {number | undefined} [shards]
  * @param {number | undefined} [shardId]
+ * @param {boolean} [color]
  */
-function writeTestConfigFile(tests, runners, light, taskConfigsFolder, workerCount, stackTraceLimit, timeout, keepFailed, shards, shardId) {
+function writeTestConfigFile(reporter, reporterOption, tests, runners, light, taskConfigsFolder, workerCount, stackTraceLimit, timeout, keepFailed, shards, shardId, color) {
     const testConfigContents = JSON.stringify({
-        test: tests ? [tests] : undefined,

```

Comment:
Be aware that removing this could possibly break individual workflows, as we were using this field in `src/testRunner/runner.ts` as an alternative to `runners` for known test runner names:

https://github.com/microsoft/TypeScript/blob/5d3284032b62748da5b8e7dd4fa8fca3ae8141a4/src/testRunner/runner.ts#L136

You may want to send an email to the team when this goes in to let them know that `gulp runtests -t fourslash` and the like won't work anymore.

Reply:
You may also want to remove `test` from the line I referenced in the comment above and the definition here:

https://github.com/microsoft/TypeScript/blob/5d3284032b62748da5b8e7dd4fa8fca3ae8141a4/src/testRunner/runner.ts#L78

Reply:
To clarify, my (wrong) guess was that this was supposed to be used only by gulp-related stuff so I looked only in `scripts/build` and missed the above completely.  (Again, see general comment.)

Reply:
`gulp runtests -t compiler` and the like are definitely used by people :(

Reply:
`-t` works the same -- it used to pass the `-g` flag to mocha, and this changes it to a `grep` option instead.

@rbuckton -- did you see some way that it would change other than the mismatch with `runner.ts`?